usageReports/reservedInstances/get_ec2_data.go

```diff
@@ -0,0 +1,122 @@
+
+	"github.com/trackit/jsonlog"
+	"github.com/trackit/trackit-server/aws/usageReports/reservedInstances"
+	"github.com/trackit/trackit-server/errors"
+	"github.com/trackit/trackit-server/es"
+	"github.com/trackit/trackit-server/users"
+)
+
+// makeElasticSearchRequest prepares and run an ES request
+// based on the reservedInstancesQueryParams and search params
+// It will return the data, an http status code (as int) and an error.
+// Because an error can be generated, but is not critical and is not needed to be known by
+// the user (e.g if the index does not exists because it was not yet indexed ) the error will
+// be returned, but instead of having a 500 status code, it will return the provided status code
+// with empty data
+func makeElasticSearchRequest(ctx context.Context, parsedParams ReservedInstancesQueryParams,
+	esSearchParams func(ReservedInstancesQueryParams, *elastic.Client, string) *elastic.SearchService) (*elastic.SearchResult, int, error) {
+	l := jsonlog.LoggerFromContextOrDefault(ctx)
+	index := strings.Join(parsedParams.IndexList, ",")
+	searchService := esSearchParams(
+		parsedParams,
+		es.Client,
+		index,
+	)
+	res, err := searchService.Do(ctx)
+	if err != nil {
+		if elastic.IsNotFound(err) {
+			l.Warning("Query execution failed, ES index does not exists", map[string]interface{}{
+				"index": index,
+				"error": err.Error(),
+			})
+			return nil, http.StatusOK, errors.GetErrorMessage(ctx, err)
+		} else if cast, ok := err.(*elastic.Error); ok && cast.Details.Type == "search_phase_execution_exception" {
+			l.Error("Error while getting data from ES", map[string]interface{}{
+				"type":  fmt.Sprintf("%T", err),
+				"error": err,
+			})
+		} else {
+			l.Error("Query execution failed", map[string]interface{}{"error": err.Error()})
+		}
+		return nil, http.StatusInternalServerError, errors.GetErrorMessage(ctx, err)
+	}
+	return res, http.StatusOK, nil
+}
+
+// GetReservedInstancesMonthly does an elastic request and returns an array of reservations monthly report based on query params
+func GetReservedInstancesMonthly(ctx context.Context, params ReservedInstancesQueryParams) (int, []ReservationReport, error) {
+	res, returnCode, err := makeElasticSearchRequest(ctx, params, getElasticSearchReservedInstancesMonthlyParams)
+	if err != nil {
+		return returnCode, nil, err
+	}
+	reservations, err := prepareResponseReservedInstancesMonthly(ctx, res)
+	if err != nil {
+		return http.StatusInternalServerError, nil, err
+	}
+	return http.StatusOK, reservations, nil
+}
+
+// GetReservedInstancesDaily does an elastic request and returns an array of daily report reservations based on query params
+func GetReservedInstancesDaily(ctx context.Context, params ReservedInstancesQueryParams, user users.User, tx *sql.Tx) (int, []ReservationReport, error) {
+	res, returnCode, err := makeElasticSearchRequest(ctx, params, getElasticSearchReservedInstancesDailyParams)
+	if err != nil {
+		return returnCode, nil, err
+	}
```

Comment:
you have to add an else if, see https://github.com/trackit/trackit-server/pull/183